gateway/requests/requests.go

```diff
@@ -90,3 +90,20 @@ type AsyncReport struct {
 type DeleteFunctionRequest struct {
 	FunctionName string `json:"functionName"`
 }
+
+// CreateSecretRequest create a secret w/ annotations
+type CreateSecretRequest struct {
+	Secret Secret `json:"secret"`
+}
+
+// DeleteSecretRequest remote a secret by name
+type DeleteSecretRequest struct {
+	SecretName string `json:"secretName"`
+}
+
+// Secret schema use Value only in write-only http verbs
+type Secret struct {
+	Name        string             `json:"name"`
+	Value       string             `json:"value"` // write-only
```

Comment:
@stefanprodan @LucasRoesler should this be `string` or `[]byte`?

Reply:
byte array makes sense to me as that opens the door save json or other structure based on that key. Vault would support it anyway.

Reply:
I am not sure if I'm right, just wanted to open discussion.

Reply:
I agree, `[]byte` seems like a good choice

Reply:
I would go with a base64 encoded string, like K8s does.

Reply:
That also makes sense. But we should make sure to include a good doc string here to make sure people remember that when they are using/implementing the secrets providers

Reply:
We should be able to represent whatever we need in a string here. Since the users won't see the value from the API we shouldn't need base64.